wherescape/connectors/gitlab/gitlab_wrapper.py

```diff
@@ -0,0 +1,198 @@
+                all_resources.append(list(final_json.values()))
+
+            try:
+                total_pages = response.headers["X-Total-Pages"]
+                current_page = response.headers["X-Page"]
+            except:
+                current_page = current_page + 1
+
+        return all_resources
+
+    def get_projects(self):
+        return self.projects
+
+    def get_projects_from_api(self):
+        keys_to_keep = COLUMN_NAMES_AND_DATA_TYPES["projects"].keys()
+        resource_api = "projects"
+
+        all_projects = self.paginate_through_resource(
+            resource_api, keys_to_keep, simple="true"
+        )
+        return all_projects
+
+    def get_release_tags(self):
+
+        keys_to_keep = COLUMN_NAMES_AND_DATA_TYPES["tags"].keys()
+
+        all_tags = []
+
+        for project in self.projects:
+            project_id = project[0]
+
+            resource_api = f"projects/{project_id}/repository/tags"
+            tag_in_tuple = self.paginate_through_resource(
+                resource_api, keys_to_keep, order_by="name", since=self.since
+            )
+
+            all_tags.extend(tag_in_tuple)
+
+        return all_tags
+
+    def get_issues(self):
+        keys_to_keep = COLUMN_NAMES_AND_DATA_TYPES["issues"].keys()
+
+        all_issues = []
+        # projects is a list of tuples, so the first item in the tuple is the id
+        for project in self.projects:
+            project_id = project[0]
+            resource_api = f"projects/{project_id}/issues"
+
+            project_issues = self.paginate_through_resource(
+                resource_api, keys_to_keep, order_by="created_at", since=self.since
+            )
+            all_issues.extend(project_issues)
+
+        return all_issues
+
+    def get_pipelines(self):
+
+        all_pipelines = []
+
+        keys_to_keep = COLUMN_NAMES_AND_DATA_TYPES["pipelines"].keys()
+        # projects is a list of tuples, so the first item in the tuple is the id
+        for project in self.projects:
+            project_id = project[0]
+            resource_api = f"projects/{project_id}/pipelines"
+            project_pipelines = self.paginate_through_resource(
+                resource_api, keys_to_keep, since=self.since
+            )
+            all_pipelines.extend(project_pipelines)
+
+        return all_pipelines
+
+    def get_merge_requests(self):
+
+        all_merge_requests = []
+
+        keys_to_keep = COLUMN_NAMES_AND_DATA_TYPES["merge_requests"].keys()
+        # projects is a list of tuples, so the first item in the tuple is the id
+        for project in self.projects:
+            project_id = project[0]
+            resource_api = f"projects/{project_id}/merge_requests"
+            project_merge_requests = self.paginate_through_resource(
+                resource_api, keys_to_keep, since=self.since, order_by="title"
+            )
+            all_merge_requests.extend(project_merge_requests)
+
+        return all_merge_requests
```

Comment:
I would have expected an `if __name__ == "__main__":` line which can be used to test this function without wherescape

Reply:
I don't think that is necessary; people can create their own 'test' files or we should create proper tests.